pkg/sql/parser/sql.y

```diff
@@ -7261,6 +7269,38 @@ simple_typename:
 | character_with_length
 | interval_type
 | postgres_oid
+| POINT error { return unimplementedWithIssueDetail(sqllex, 21286, "point") } // needed or else it generates a syntax error.
```

Comment:
What kind of syntax are you trying to catch here?

Reply:
because we added POINT to the keywords, it doesn't catch `CREATE TABLE TYPE (a POINT)` like we used to (somewhere deeper in sql layer), so moving that up here.